modules/aws-eks-self-managed-node-groups/main.tf

```diff
@@ -5,9 +5,45 @@ resource "aws_autoscaling_group" "self_managed_ng" {
   min_size            = local.self_managed_node_group["min_size"]
   vpc_zone_identifier = length(local.self_managed_node_group["subnet_ids"]) == 0 ? (local.self_managed_node_group["subnet_type"] == "public" ? var.context.public_subnet_ids : var.context.private_subnet_ids) : local.self_managed_node_group["subnet_ids"]
 
-  launch_template {
-    id      = module.launch_template_self_managed_ng.launch_template_id[local.lt_self_managed_group_map_key]
-    version = module.launch_template_self_managed_ng.launch_template_latest_version[local.lt_self_managed_group_map_key]
+  capacity_rebalance = local.self_managed_node_group["capacity_rebalance"]
+
+  dynamic "launch_template" {
```

Comment:
this is now a breaking change so we'll have to figure out what our plan is - cc @askulkarni2 / @kcoleman731

Reply:
I'm happy to discuss this @bryantbiggs, the examples I uploaded are using only one instance type (as the rest of the examples in this repo) to test that path to make sure it's backwards compatible.